services/director-v2/src/simcore_service_director_v2/modules/dynamic_sidecar/docker_api.py

```diff
@@ -506,3 +511,39 @@ async def try_to_remove_network(network_name: str) -> None:
             return await network.delete()
         except aiodocker.exceptions.DockerError:
             log.warning("Could not remove network %s", network_name)
+
+
+async def update_scheduler_data_label(scheduler_data: SchedulerData) -> None:
+    async with docker_client() as client:
+        # NOTE: builtin `DockerServices.update` function is very limited.
+        # Using the same pattern but updating labels
+
+        try:
+            # fetch information from service name
+            service_inspect = await client.services.inspect(scheduler_data.service_name)
+            service_version = service_inspect["Version"]["Index"]
+            service_id = service_inspect["ID"]
+            spec = service_inspect["Spec"]
+
+            # allows to use json encodes not available on dict
+            dict_scheduler_data = json.loads(scheduler_data.json())
+            # compose_spec needs to be json encoded
+            dict_scheduler_data["compose_spec"] = json.dumps(
+                dict_scheduler_data["compose_spec"]
+            )
+            label_data = json.dumps(dict_scheduler_data)
+            spec["Labels"][DYNAMIC_SIDECAR_SCHEDULER_DATA_LABEL] = label_data
```

Comment:
I do not understand this. Are you modifying here the image labels?

Reply:
Yes I am overwriting a label, the one which keeps track of the state of the service.